src/app/shared/components/cities-search/display-location-option.ts

```diff
@@ -20,8 +20,9 @@ export function displayLocationOption(location?: Location) {
   return toTitleCase(location.city) + ', ' + location.region;
 
   function toTitleCase(value: string) {
-    // DO NOT replace it with `value[0].toUpperCase()` etc. as that may not
-    // select the whole glyph, unlike unicode regex replacement.
-    return value.toLowerCase().replace(/^./u, (letter) => letter.toUpperCase());
+    // ^. - first character of the string
+    // \s. - first character after whitespace
+    // -. - first character after - (f.e. Kudowa-Zdroj)
+    return value.toLowerCase().replace(/(^.|\s.|-.)([^-\s]*)/g, (_, m1, m2) => m1.toUpperCase() + m2);
```

Comment:
Please use the `u` flag to make sure Unicode glyphs are handled correctly, just in case.
```suggestion
    return value.toLowerCase().replace(/(^.|\s.|-.)([^-\s]*)/gu, (_, m1, m2) => m1.toUpperCase() + m2);
```